index.bs

```diff
@@ -247,6 +247,202 @@ a generator to a processor (e.g., with a call like
 the Web Platform-exposed signals, but ignores any internal custom signals as there is no
 way for the platform sinks to know what the upstream source is.
 
+## MediaStreamTrack ## {#generator-as-track}
+A {{MediaStreamTrackGenerator}} is a {{MediaStreamTrack}}. This section adds
+clarifications on how a {{MediaStreamTrackGenerator}} behaves as a
+{{MediaStreamTrack}}.
+
+### clone ### {#generator-clone}
+The {{MediaStreamTrack/clone}} method on a {{MediaStreamTrackGenerator}}
+returns a new {{MediaStreamTrack}} object whose underlying source is the
+same as the one for the {{MediaStreamTrackGenerator}} being cloned.
+This underlying source is the {{MediaStreamTrackGenerator/writable}} of
+the {{MediaStreamTrackGenerator}}.
+
+### stop ### {#generator-stop}
+The {{MediaStreamTrack/stop}} method on a {{MediaStreamTrackGenerator}} stops
+the track. When the last track connected to
+the {{MediaStreamTrackGenerator/writable}} of a {{MediaStreamTrackGenerator}} is
+stopped, the {{MediaStreamTrackGenerator/writable}} is closed.
+
+### Constraintable properties ### {#generator-constrainable-properties}
+
+The following constrainable properties are defined for video
+{{MediaStreamTrackGenerator}}s and any {{MediaStreamTrack}}s sourced from
+a {{MediaStreamTrackGenerator}}:
+<table>
+  <thead>
+    <tr>
+      <th>
+        Property Name
+      </th>
+      <th>
+        Values
+      </th>
+      <th>
+        Notes
+      </th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr id="def-constraint-width">
+      <td data-tests="">
+        <dfn>width</dfn>
+      </td>
+      <td>
+        {{ConstrainULong}}
+      </td>
+      <td>
+        As a setting, this is the width, in pixels, of the latest
+        frame delivered by the track to its sinks. As a capability, `max` MUST reflect the
+        largest width a {{VideoFrame}} may have, and `min`
+        MUST reflect the smallest width a {{VideoFrame}} may have.
+      </td>
+    </tr>
+    <tr id="def-constraint-height">
+      <td data-tests="">
+        <dfn>height</dfn>
+      </td>
+      <td>
+        {{ConstrainULong}}
+      </td>
+      <td>
+        As a setting, this is the height, in pixels, of the latest
+        frame delivered by the track to its sinks. As a capability, `max` MUST reflect the largest height
+        a {{VideoFrame}} may have, and `min` MUST reflect
+        the smallest height a {{VideoFrame}} may have.
+      </td>
+    </tr>
+    <tr id="def-constraint-frameRate">
+      <td data-tests="">
+        <dfn>frameRate</dfn>
+      </td>
+      <td>
+        {{ConstrainDouble}}
+      </td>
+      <td>
+        As a setting, this is an estimate of the frame rate based on frames
+        recently delivered by the track to its sinks. As a capability `min` MUST be zero and
```

Comment:
This is the same definition used by webrtc-pc. The only difference is that this one uses frames delivered by the track instead of frames received by the track because constraints may affect the settings of output frames of individual tracks.